pkg/presidio/templates/templates.go

```diff
@@ -28,18 +35,41 @@ func createKey(project string, action string, id string) string {
 // GetTemplate from key store
 func (templates *Templates) GetTemplate(project string, action string, id string) (string, error) {
 	key := createKey(project, action, id)
+	if templates.cacheStore != nil {
+		res, err := templates.cacheStore.Get(key)
```

Comment:
if the key is not in the cache, but the cache is configured, it should be added to the cache..